components/SponsoredFeedsTable.tsx

```diff
@@ -0,0 +1,195 @@
+import { useState } from "react";
+import CopyIcon from "./icons/CopyIcon";
+
+interface SponsoredFeed {
+  name: string;
+  priceFeedId: string;
+  updateParameters: string;
+}
+
+interface SponsoredFeedsTableProps {
+  feeds: SponsoredFeed[];
+  networkName: string;
+}
+
+export const SponsoredFeedsTable = ({
+  feeds,
+  networkName,
+}: SponsoredFeedsTableProps) => {
+  const [copiedId, setCopiedId] = useState<string | null>(null);
+
+  const copyToClipboard = (text: string) => {
+    navigator.clipboard.writeText(text).then(() => {
+      setCopiedId(text);
+      setTimeout(() => setCopiedId(null), 2000);
+    });
+  };
+
+  // Calculate parameter statistics
+  const paramCounts = feeds.reduce((acc, feed) => {
+    acc[feed.updateParameters] = (acc[feed.updateParameters] || 0) + 1;
+    return acc;
+  }, {} as Record<string, number>);
+
+  const defaultParams = Object.entries(paramCounts).sort(
+    ([, a], [, b]) => b - a
+  )[0][0];
+
+  // Calculate table height based on number of items
+  // Each row is approximately 40px (py-2 = 8px top + 8px bottom + content height)
+  // Header is approximately 48px (py-2 = 8px top + 8px bottom + font height)
+  // Show 7 rows by default, then scroll - but maintain consistent minimum height
+  const maxVisibleRows = 7;
+  const shouldScroll = feeds.length > maxVisibleRows;
+  const rowHeight = 56; // Increased row height to account for actual content height
+  const headerHeight = 48; // Header height in pixels
+  const exactTableHeight = `${headerHeight + maxVisibleRows * rowHeight}px`; // Exact height for 7 rows
+  const tableHeight = shouldScroll ? exactTableHeight : "auto"; // Use exact height for scrollable tables
```

Comment:
I really dislike this kind of thing, it's very brittle (i.e. it's a pain to ensure every change is tested across device sizes and it's easy to do something that changes the calculations and not know it breaks the UI) and it's very easy to run into render issues on different platforms where size calculations have slight differences.

It also feels unnecessarily complex for what you're looking to accomplish.

In this case, what I'd suggest doing is just adding the `max-height` css property to the table

Reply:
yep I think this makes more sense to me